src/platform_impl/macos/window_delegate.rs

```diff
@@ -821,7 +824,15 @@ impl WindowDelegate {
     }
 
     pub fn set_transparent(&self, transparent: bool) {
-        self.window().setOpaque(!transparent)
+        self.window().setOpaque(!transparent);
+
+        let color = if transparent {
+            unsafe { NSColor::clearColor() }
+        } else {
+            unsafe { NSColor::windowBackgroundColor() }
+        };
+
+        self.window().setBackgroundColor(Some(color).as_deref());
```

Comment:
Nit: Slightly clearer.

```suggestion
        self.window().setBackgroundColor(Some(&color));
```